src/chunk.c

```diff
@@ -1860,3 +1845,52 @@ chunks_return_srf(FunctionCallInfo fcinfo)
 	else						/* do when there is no more left */
 		SRF_RETURN_DONE(funcctx);
 }
+
+static void
+drop_chunks_main(Oid table_relid, Datum older_than_datum, Datum newer_than_datum, Oid older_than_type, Oid newer_than_type, bool cascade)
+{
+	int			i = 0;
+	uint64		num_chunks = 0;
+	Chunk	  **chunks = show_drop_chunks(table_relid, older_than_datum, newer_than_datum, older_than_type, newer_than_type, "drop_chunks", CurrentMemoryContext, &num_chunks);
+
+	for (; i < num_chunks; i++)
+	{
+		ObjectAddress objaddr = {
+			.classId = RelationRelationId,
+			.objectId = chunks[i]->table_id,
+		};
+
+		/* Remove the chunk from the hypertable table */
+		chunk_delete_by_relid(chunks[i]->table_id);
+
+		/* Drop the table */
+		performDeletion(&objaddr, cascade, 0);
+	}
+}
+
+Datum
+ts_chunk_drop_chunks(PG_FUNCTION_ARGS)
+{
+	Oid			table_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Datum		older_than_datum = PG_GETARG_DATUM(1);
+	Datum		newer_than_datum = PG_GETARG_DATUM(2);
+
+	/*
+	 * get_fn_expr_argtype defaults to UNKNOWNOID if argument is NULL but
+	 * making it InvalidOid makes the logic simpler later
+	 */
+	Oid			older_than_type = PG_ARGISNULL(1) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 1);
+	Oid			newer_than_type = PG_ARGISNULL(2) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 2);
+	bool		cascade = PG_GETARG_BOOL(3);
+
+	drop_chunks_main(table_relid, older_than_datum, newer_than_datum, older_than_type, newer_than_type, cascade);
+
+	PG_RETURN_NULL();
+}
+
+/* A main function that can be called by the scheduler */
+void
+drop_chunks_main_wrapper(Oid table_relid, Interval *older_than, bool cascade)
```

Comment:
does this have to be here? Can't it be in scheduler code. can't we just expose `drop_chunks_main` (with another name)?

Reply:
Yeah, I considered that too. I'm just going to remove it from this PR and add it elsewhere.

Reply:
Yeah, I considered that too. I'm just going to remove it from this PR and add it elsewhere.